src/rust/engine/dep_inference/Cargo.toml

```diff
@@ -1,5 +1,14 @@
 [package]
-version = "0.0.2"
+# NB: The Major/Minor is loosely tied to Pants version of the latest breaking change.
+# The patch version should be incremented every time a breaking change is cherry-picked.
+# So when you're making a change on `main`:
+# - Set this to main's Major/Minor.0 if it isn't already
+# - Otherwise, increment the patch number
+# Then when/if you cherry-pick and it fails, simply increment the patch number on the target branch.
+#
+# (A "breaking change" is one that changes the behavior of dependency parsing. This version is
+# embedded in the local cache. Therefore if behavior changes, the cache should be busted)
+version = "2.18.0"
```

Comment:
Does this mean we need to bump this version manually each time we branch? What do you think about just incorporatin Pants' major/minor into the cache key directly and sticking with the old style?

Reply:
I'm hoping the long tail will settle such that multiple releases don't require cache busting.

And that's the intent here. We only change the version if it's needed, but we're doing so that's intentionally forcing a choice to be made in cherry-pick. Because otherwise the version bump picks cleanly, which is dangerous.

So, eventually without changes, the version here could be `2.20.0` and we've gone through 6 releases so main is at `2.26.0` and each of those releases continue to reuse the cache that was last busted in `2.20.0`.

So I guess think of it that way. Major/minor "version" is the last time the cache was busted.

Reply:
Ah okay, makes sense. Thanks. 👍